src/backend/distributed/commands/sequence.c

```diff
@@ -213,6 +213,46 @@ ExtractDefaultColumnsAndOwnedSequences(Oid relationId, List **columnNameList,
 }
 
 
+/*
+ * ColumnDefaultsToNextVal returns true if the column with attrNumber
+ * has a default expression that contains nextval().
+ */
+bool
+ColumnDefaultsToNextVal(Oid relationId, AttrNumber attrNumber)
+{
+	Relation relation = RelationIdGetRelation(relationId);
+
+	TupleDesc relTupDesc = RelationGetDescr(relation);
+	TupleConstr *relTupConstraints = relTupDesc->constr;
+	if (relTupConstraints == NULL)
+	{
+		/* none of the columns have a DEFAULT expression */
+		RelationClose(relation);
+		return false;
+	}
+
+	for (uint16 defExprIndex = 0; defExprIndex < relTupConstraints->num_defval;
+		 defExprIndex++)
+	{
```

Comment:
can we use `build_column_default ` function here? It is kind of tricky to deal with these internal data structures.

(similar thing applies to #6340 as well)

Reply:
Wow, I didn't know this function :) Will use this.